packages/fxa-settings/src/pages/ResetPassword/ResetPasswordRecoveryChoice/container.tsx

```diff
@@ -99,7 +100,43 @@ export const ResetPasswordRecoveryChoiceContainer = (
       }
       return AuthUiErrors.UNEXPECTED_ERROR;
     }
-  };
+  }, [authClient, locationState, navigateWithQuery]);
+
+  // Auto send code and navigate if only phone available (0 backup codes)
+  useEffect(() => {
+    if (
+      autoSendAttempted ||
+      loading ||
+      numBackupCodes > 0 ||
+      !phoneData.phoneNumber ||
+      !locationState?.state?.token
+    ) {
+      return;
+    }
+
+    setAutoSendAttempted(true);
+    (async () => {
+      const error = await handlePhoneChoice();
+      navigateWithQuery('/reset_password_recovery_phone', {
+        state: {
+          ...locationState.state,
+          lastFourPhoneDigits: phoneData.lastFourPhoneDigits,
+          ...(error ? { sendError: error } : {}),
+          numBackupCodes,
+        },
+        replace: true,
+      });
```

Comment:
non-blocking comment/question: when we navigate away while in the component scope, we return early so that the rest of the code won't execute.  Do we need to do something similar?

Reply:
I don't think so. The synchronous navigations use early returns to prevent the component from rendering anything, but this async navigation in the useEffect is different - it runs as a side effect while the component shows a loading spinner. The navigation will eventually take the user away and React will clean up automatically. An early return here wouldn't really do anything.

Reply:
Ah, I shouldn't have taken your comment quite so literally. I've refactored to avoid having concurrent useEffects that might unnecessarily execute.